src/hotspot/share/runtime/threads.cpp

```diff
@@ -648,6 +650,8 @@ jint Threads::create_vm(JavaVMInitArgs* args, bool* canTryAgain) {
   }
 
   initialize_java_lang_classes(main_thread, CHECK_JNI_ERR);
+  // Update the java.vm.vendor property with java.lang.VersionProps.VM_VENDOR
+  Arguments::update_vm_vendor(VM_Version::vm_vendor());
```

Comment:
It seems very ad hoc to update this after initPhase1 has run, need to think through if there is a better way to do this.

Reply:
I agree. This is to minimize change on behavior. 

Not sure what kind of requirement is there for java.vm.vendor, would it be OK for VM not the set this value and wait until loading of VersionProps class?

The update is needed for the assertion in statSampler, I don't see other reference to the property in the repo. So the question is when we need the property to be available and exposure scope.

Reply:
I can do it together, override_vm_vendor read the value from the VersionProps class and set the value in VM. 
update_vm_vendor update the java.vm.vendor system property with the override value.

So I think override is part of the initialize class, and the replace the system property after initialize of VersionProps.